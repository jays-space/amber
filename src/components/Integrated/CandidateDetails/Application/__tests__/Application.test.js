import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Application } from "..";

describe("<Application />", () => {
  const mockRegisteredData = {
    date: "2016-09-28T11:15:47.538Z",
    age: 5,
  };

  const mockIDData = {
    name: "AVS",
    value: "756.9989.7526.17",
  };

  it("should render correctly", () => {
    render(<Application registered={mockRegisteredData} id={mockIDData} />);

    const applicationComponent = screen.getByTitle("about-application");
    expect(applicationComponent).toBeInTheDocument();
    expect(applicationComponent).toMatchSnapshot();
  });

  it("should render appropriate heading", () => {
    render(<Application registered={mockRegisteredData} id={mockIDData} />);

    const headingElement = screen.getByRole("heading", { name: "application" });
    expect(headingElement).toBeInTheDocument();
  });

  it("should render registration details", () => {
    render(<Application registered={mockRegisteredData} id={mockIDData} />);

    const registrationComponents = screen.getAllByTestId(
      "registration-details"
    );
    expect(registrationComponents.length).toBeGreaterThan(1);
  });

  it("should render identification details", () => {
    render(<Application registered={mockRegisteredData} id={mockIDData} />);

    const identificationComponent = screen.getByTestId(
      "identification-details"
    );
    expect(identificationComponent).toBeInTheDocument();
  });
});
