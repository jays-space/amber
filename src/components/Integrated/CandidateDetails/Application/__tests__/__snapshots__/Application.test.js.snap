// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application /> should render correctly 1`] = `
<div
  title="about-application"
>
  <h4
    class="mb-6 lg:mb-2 text-slate-700 capitalize text-xl font-bold"
  >
    application
  </h4>
  <div
    class="md:p-4 grid grid-cols-1 md:grid-cols-2 gap-6 md:border rounded-md"
  >
    <div
      class="flex flex-col hidden md:grid"
      data-testid="registration-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        registration
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          Wed, 28 Sept 2016
        </p>
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          (5 years ago)
        </p>
      </span>
    </div>
    <div
      class="grid gap-6 md:hidden"
    >
      <div
        class="flex flex-col "
        data-testid="registration-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          registration
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            Wed, 28 Sept 2016
          </p>
        </span>
      </div>
      <div
        class="flex flex-col "
        data-testid="registration-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          registration
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            (5 years ago)
          </p>
        </span>
      </div>
    </div>
    <div
      class="flex flex-col "
      data-testid="identification-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        identification
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          AVS
        </p>
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          - 756.9989.7526.17
        </p>
      </span>
    </div>
  </div>
</div>
`;
